contracts/PercentageTransferManager/src/lib.rs

```diff
@@ -0,0 +1,520 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+mod custom_types {
+
+    use ink_core::storage::Flush;
+    use scale::Encode;
+
+    #[derive(
+        scale::Decode, scale::Encode, PartialEq, Ord, Eq, PartialOrd, Copy, Hash, Clone, Debug,
+    )]
+    #[cfg_attr(feature = "ink-generate-abi", derive(type_metadata::Metadata))]
+    pub struct IdentityId([u8; 32]);
```

Comment:
Couldn't we use the `IdentityId` defined in `polymesh-primitives`? We could extend it to implement `Flush`, `Metadata`, etc.

Reply:
Initially, I was trying to do the same but ink! doesn't allow me to access those. I think it is more of a dependency issue on the ink side that we can resolve in another task if possible